cloudferrylib/os/actions/snapshots.py

```diff
@@ -27,14 +27,18 @@ def run(self, *args, **kwargs):
         # dump mysql to file
         # probably, we have to choose what databases we have to dump
         # by default we dump all databases
+        user = self.cloud.cloud_config.mysql.user
+        password = self.cloud.cloud_config.mysql.password
+        path = self.cloud.cloud_config.snapshot.snapshot_path
+        if password == '':
+            password_arg = ''
+        else:
+            password_arg = "--password=" + password
         command = ("mysqldump "
-                   "--user={user} "
-                   "--password={password} "
+                   "--user={0} "
+                   "{1} "
                    "--opt "
-                   "--all-databases > {path}").format(
-            user=self.cloud.cloud_config.mysql.user,
-            password=self.cloud.cloud_config.mysql.password,
-            path=self.cloud.cloud_config.snapshot.snapshot_path)
+                   "--all-databases > {2}").format(user, password_arg, path)
```

Comment:
Changing `{user}` and `{password}` to `{0}` and `{1}` is less readable than it used to be, please revert